packages/runtime-dom/src/apiCustomElement.ts

```diff
@@ -263,7 +264,7 @@ export class VueElement
         )
       }
       if (_def.shadowRoot !== false) {
-        this.attachShadow({ mode: 'open' })
+        this.attachShadow(extend({ mode: 'open' }, _def.shadowRootOptions))
```

Comment:
```suggestion
        this.attachShadow(extend({}, _def.shadowRootOptions,{ mode: 'open' }))
```
to avoid `mode` being overwritten

Reply:
What if I wanted to override the mode? Just like other options it'd be good to at least have the option to override them. Is there a reason why the mode needs to be `open`?

Reply:
It might cause some bugs, but I haven’t tested it thoroughly.

Reply:
I just did - it looks like `mode: 'closed'` renders the component completely unusable. Changing as requested, but then I'll exclude the `mode` option from `shadowRootOptions` to signify it.

Reply:
Done